src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -21,11 +21,23 @@ namespace ILCompiler
 {
     public class CompilationOptions
     {
-        public IReadOnlyDictionary<string, string> InputFilePaths;
-        public IReadOnlyDictionary<string, string> ReferenceFilePaths;
+        /// <summary>
+        /// Associate for each simple name, the list of assembly full path names matching the simple name.
+        /// Useful when including multiple assemblies with same name from different locations.
```

Comment:
This should be a very simple check in AppendExpendedPaths. The code for it is there already - it just does the enforcement for the subject of compilation, but for references today.